test/integration/buckets-client-validation.test.js

```diff
@@ -147,6 +186,62 @@ test('buckets client validation', testOpts, function (suite) {
         });
     });
 
+    VALID_LIST_BUCKETS_QUERY_OPTS.forEach(function (query_opts) {
+        test(f('validating list buckets: %j', query_opts), function (t) {
+            var s = client.createListBucketsStream({query: query_opts});
+
+            s.on('readable', function onReadable() {
+                while (s.read() !== null) {
+                    // exhaust the stream
+                }
+            });
+
+            s.once('error', function onError(err) {
+                t.ifError(err);
+                t.end();
+            });
+
+            s.once('end', function onEnd() {
+                t.ok('end seen');
```

Comment:
Were you expecting that "end seen" to be a test case message?  Instead that is testing that `end seen` is truthy:

```
...
    # Subtest: validating list buckets: {}
        ok 1 - expect truthy value
        1..1
    ok 19 - validating list buckets: {} # time=59.918ms

    # Subtest: validating list buckets: {"limit":100}
        ok 1 - expect truthy value
        1..1
    ok 20 - validating list buckets: {"limit":100} # time=57.108ms

    # Subtest: validating list buckets: {"limit":1,"delimiter":"/"}
        ok 1 - expect truthy value
        1..1
    ok 21 - validating list buckets: {"limit":1,"delimiter":"/"} # time=58.686ms
...
```

Ditto for other `t.ok('end seen');` below.

Reply:
ah yes. i originally had this as a `console.log` for testing and incorrectly converted it to a test - good catch